otcextensions/tests/functional/sdk/dns/v2/test_zone.py

```diff
@@ -29,12 +29,13 @@ def setUp(self):
         super(TestZone, self).setUp()
 
     def tearDown(self):
-        try:
-            if self.zone:
+        if self.zone:
+            try:
                 self.client.delete_zone(self.zone)
-        except openstack.exceptions.SDKException as e:
-            _logger.warning('Got exception during clearing resources %s'
-                            % e.message)
+            except openstack.exceptions.SDKException as e:
+                _logger.warning('Got exception during clearing resources %s'
+                                % e.message)
+            self.client.wait_for_delete_zone(self.zone)
```

Comment:
I would rather put wait_for_delete also under the try, cause here we can get timeout

Reply:
fixed